### PR TITLE
Added support for Fixie test framework

### PR DESCRIPTION
**Pull request by ChrisMissal**
Fixie - Conventional Testing for .NET

This adds the bare minimum support for executing tasks with Fixie.

    Task("Tests").Does(() => {
        Fixie(@".\path\tests.dll");
    });

### CODE REVIEW

**Comment by ChrisMissal** [comment]
This is a work in progress for #281

**Comment by patriksvensson** [comment]
Looking good! Nice! :+1:

**Comment by ChrisMissal** [comment]
@patriksvensson Fixie allows for any named Options as arguments like so:

```
Fixie.Console.exe path/to/your/test/project.dll --include CategoryA --include CategoryB
```

Is there a precedent for this anywhere else that I can follow?

**Comment by ChrisMissal** [comment]
These two approaches both work:

**As a collection of KeyValuePairs**
```csharp
runner.Run("./Tests.dll", new FixieSettings
{
    Options = new Collection<KeyValuePair<string, string>>
    {
        new KeyValuePair<string, string>("--include", "CategoryA"),
        new KeyValuePair<string, string>("--include", "CategoryB"),
    }
});
```

and

**As a string array**
```csharp
runner.Run("./Tests.dll", new FixieSettings
{
    Options = new[] { "--include", "CategoryA", "--include", "CategoryB" },
});
```
/cc @plioi

**Comment by patriksvensson** [comment]
The only tool I can think of that does the same is the MSBuild alias that exposes an `IDictionary<string, IList<string>>` for properties. To make it easier to work with it we expose a fluent API for the settings.

Would be cool with a fluent API on top of `FixieSettings`, but that's just my opinion. I think the important thing is that it's easy(ish) to use without intellisense.

```csharp
runner.Run("./Tests.dll", new FixieSettings()
   .WithOption("--include", "CategoryA")
   .WithOption("--include", "CategoryB"));
```

**Comment by plioi** [comment]
FWIW internally, and not surprisingly, Fixie also represents these as an IDictionary<string, List<string>>

**Comment by ChrisMissal** [comment]
Thanks for the quick responses! I went with the `.WithOption` approach @patriksvensson, good call!

I think this should do it for adding Fixie, all tests are good, and I can run my build.csx file against other, real-life projects. :smile:

File: src/Cake.Common/Tools/Fixie/FixieAliases.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+using Cake.Core;
+using Cake.Core.Annotations;
+using Cake.Core.IO;
+
+namespace Cake.Common.Tools.Fixie
+{
+    /// <summary>
+    /// Contains functionality related to running Fixie tests.
+    /// </summary>
+    [CakeAliasCategory("NUnit")]
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/Fixie/FixieAliases.cs]: Should be `Fixie`.

File: src/Cake.Common.Tests/Unit/Tools/Fixie/FixieRunnerTests.cs [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+                // Then
+                fixture.ProcessRunner.Received(1).Start(
+                    Arg.Any<FilePath>(),
+                    Arg.Is<ProcessSettings>(p =>
+                        p.Arguments.Render() == "\"/Working/blarg.dll\" --NUnitXml \"/Working/nunit-style-results.xml\""));
+            }
+
+            [Fact]
+            public void Should_set_xUnitXml_Output_File()
```
Comment (patriksvensson) [inline src/Cake.Common.Tests/Unit/Tools/Fixie/FixieRunnerTests.cs]: Minor, but should be `Should_Set_xUnitXml_Output_File` :smile:

File: src/Cake.Common.Tests/Unit/Tools/Fixie/FixieRunnerTests.cs [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+                // Then
+                fixture.ProcessRunner.Received(1).Start(
+                    Arg.Any<FilePath>(),
+                    Arg.Is<ProcessSettings>(
+                        p => p.Arguments.Render() == string.Format("\"/Working/Tests.dll\" --TeamCity {0}", teamCityValue)));
+            }
+
+            [Fact]
+            public void Should_Set_custom_options()
```
Comment (patriksvensson) [inline src/Cake.Common.Tests/Unit/Tools/Fixie/FixieRunnerTests.cs]: Should be `Should_Set_Custom_Options()` (uppercase)

File: src/Cake.Common.Tests/Unit/Tools/Fixie/FixieRunnerTests.cs [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+                fixture.ProcessRunner.Received(1).Start(
+                    Arg.Any<FilePath>(),
+                    Arg.Is<ProcessSettings>(p =>
+                        p.Arguments.Render() == "\"/Working/blarg.dll\" --xUnitXml \"/Working/xunit-results.xml\""));
+            }
+
+            [Theory]
+            [InlineData(TeamCityOutput.On, "on")]
+            [InlineData(TeamCityOutput.Off, "off")]
+            public void Should_Set_TeamCity_value(TeamCityOutput? teamCityOutput, string teamCityValue)
```
Comment (patriksvensson) [inline src/Cake.Common.Tests/Unit/Tools/Fixie/FixieRunnerTests.cs]: Should be `Should_Set_TeamCity_Value()` (uppercase)

**Comment by patriksvensson** [comment]
@ChrisMissal Left some comments for you. Sorry about the nitpicking about the casing :smile: 

Another thing I was thinking of is if the options should be `IDictionary<string, IList>` to follow the MSBuild conventions. What do you think?

Otherwise it looks good to me. Great work! :+1:

**Comment by ChrisMissal** [comment]
No problem, I'll make the necessary changes.

@patriksvensson Do you want the WithOption to also be an extension method like the [MSBuildSettingsExtensions.WithProperty](https://github.com/cake-build/cake/blob/95888803e676a5e0c61a2c4f8b64d0aa9cf8950f/src/Cake.Common/Tools/MSBuild/MSBuildSettingsExtensions.cs#L68)?

**Comment by patriksvensson** [comment]
@ChrisMissal I don't mind either way. Moving it to an extension method in the same namespace later on would not break anything so it's up to you :smile:

**Comment by ChrisMissal** [comment]
@patriksvensson Alright, changes made!

**Comment by patriksvensson** [comment]
@ChrisMissal Perfect!

NuGet seem to have some problems at the moment (that's why the build fails) and it's 02:00 AM here, so I'll merge everything tomorrow and it will be available in the next release (hopefully tomorrow).

**Comment by patriksvensson** [comment]
Merged :+1:

**Pull request closed without merging**